test/ref_ops_nonstd_shape_test.cpp

```diff
@@ -55,4 +55,62 @@ TEST_CASE(argmin_test_nonstd_shape)
     EXPECT(migraphx::verify_range(result_vec, res_gold_vec));
 }
 
+TEST_CASE(squeeze_transpose_test)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    std::vector<float> data(4 * 3 * 3);
+    migraphx::shape s1{migraphx::shape::float_type, {4, 1, 3, 1, 3}};
+    auto l0 = mm->add_literal(migraphx::literal{s1, data});
+    auto l0_trans =
+        mm->add_instruction(migraphx::make_op("transpose", {{"permutation", {1, 2, 3, 0, 4}}}), l0);
+    mm->add_instruction(migraphx::make_op("squeeze"), l0_trans);
+    auto p_uncompiled = p;
+    p.compile(migraphx::ref::target{});
+    auto result         = p.eval({}).back();
+    auto* mm_uncompiled = p_uncompiled.get_main_module();
+    // ref implementation always does the auto_contiguous pass
+    migraphx::run_passes(*mm_uncompiled, {migraphx::auto_contiguous{}});
```

Comment:
I think adding a contiguous here does not test what you want. In this case, the output of the uncompiled program is a transposed shape, you need to manually do a transpose for the output according to the remaining dimension in the above transpose op (permutation is {1, 2, 0}).  and compared the output to that of the compiled program.

Reply:
`contiguous` will preserve order of elements but does not preserve memory layout. However, this test only checks the shape and not the values in the tensor, which is a problem.

Reply:
If you add `auto_contiguous` then you need to add `eliminate_contiguous` to make sure it wont remove a `contiguous` thats needed. Although, we shouldn't be running `auto_contiguous` on the uncompiled version.

Reply:
@scxiao @pfultz2 I've added a contiguous op instead of using `auto_contiguous` pass.